org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/jvmmodel/JvmModelAssociator.java

```diff
@@ -467,6 +467,9 @@ public void removeAssociation(EObject sourceElement, EObject jvmElement) {
 		if (sources != null && sources.remove(sourceElement)) {
 			Set<EObject> targets = sourceToTargetMap(resource).get(sourceElement);
 			targets.remove(jvmElement);
```

Comment:
What if targets is empty, now? Should it be removed from the sourceToTargetMap, too?

Reply:
did not check if that would have any side effects

Reply:
Ok